### ISSUE TITLE
`no-unknown-classes`: `strings` matcher flags string literals passed to nested function calls inside `clsx`/`cn`

### ISSUE THREAD

**Issue by volker-attempto** [schoero/eslint-plugin-readable-tailwind#385]
## Description

`better-tailwindcss/no-unknown-classes` reports string literals that are arguments to a **nested function call** inside a callee (e.g. `clsx`), even though those strings are not class names.

The default matcher for `clsx`/`cn` is `{ match: "strings" }`. Per the [advanced configuration docs](https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/main/docs/configuration/advanced.md), the `strings` matcher should collect **only direct string literals** — strings inside nested structures are supposed to require explicit matchers (`objectValues`, `anonymousFunctionReturn`, …). But a string passed to a nested `CallExpression` (which is neither a direct argument nor an object value / anonymous-function return) is still collected and linted.

This appears to be the same class of over-matching as the previously fixed #226 and #210 (under the rule's former name `no-unregistered-classes`), but for nested call arguments specifically. I couldn't find an existing open issue for this case.

### Flavor

TSX

### Code Input

```tsx
import clsx from "clsx";

// `helper`'s argument is a mode string, NOT a class name
const helper = (mode: string) => (mode === "a" ? "flex" : "block");

const Component = () => (
  <div className={clsx(helper("not-a-classname"), "flex")}>x</div>
);
```

### Expected Behavior

No error. "not-a-classname" is an argument to `helper`, not a direct string literal argument of clsx(), so the `strings` matcher should ignore it.
(Only "flex" should be considered a class.)

### Actual Behavior

```tsx
"not-a-classname" is reported as an unknown class:


//   Unknown class detected: not-a-classname  better-tailwindcss/no-unknown-classes
```

### ESLint Log

```shell
/repro/Component.tsx
  6:24  error  Unknown class detected: not-a-classname  better-tailwindcss/no-unknown-classes

✖ 1 problem (1 error, 0 warnings)
```

### ESLint Config

```typescript
import betterTailwind from "eslint-plugin-better-tailwindcss";

export default [
  {
    files: ["**/*.{ts,tsx}"],
    extends: [betterTailwind.configs.correctness],
    settings: {
      "better-tailwindcss": {
        entryPoint: "src/styles/globals.css",
      },
    },
  },
];
```

### Versions

- ESLint: 9.39.4
- Parser: typescript-eslint (flat config)
- Plugin: eslint-plugin-better-tailwindcss 4.6.0
- Node: 24.12.0
- Tailwind CSS: 4.3.1
